packages/gatsby/src/recipes/cli.js

```diff
@@ -0,0 +1,434 @@
+const fs = require(`fs`)
+const path = require(`path`)
+
+const React = require(`react`)
+const { useState, useContext, useEffect } = require(`react`)
+const { render, Box, Text, useInput, useApp } = require(`ink`)
+const Spinner = require(`ink-spinner`).default
+const unified = require(`unified`)
+const remarkMdx = require(`remark-mdx`)
+const remarkParse = require(`remark-parse`)
+const remarkStringify = require(`remark-stringify`)
+const jsxToJson = require(`simplified-jsx-to-json`)
+const visit = require(`unist-util-visit`)
+const humanizeList = require(`humanize-list`)
+const {
+  createClient,
+  useMutation,
+  useSubscription,
+  Provider,
+  defaultExchanges,
+  subscriptionExchange,
+} = require(`urql`)
+const { SubscriptionClient } = require(`subscriptions-transport-ws`)
+const fetch = require(`node-fetch`)
+const ws = require(`ws`)
+
+const isRelative = path => {
+  if (path.slice(0, 1) == `.`) {
+    return true
+  }
+
+  return false
+}
+
+module.exports = ({ recipe, projectRoot }) => {
+  let recipePath
+  if (isRelative(recipe)) {
+    recipePath = path.join(projectRoot, recipe)
+  } else {
+    recipePath = path.join(__dirname, recipe)
+  }
+  if (recipePath.slice(-4) !== `.mdx`) {
+    recipePath += `.mdx`
+  }
+  const recipeSrc = fs.readFileSync(recipePath, `utf8`)
+
+  const GRAPHQL_ENDPOINT = `http://localhost:4000/graphql`
+
+  const subscriptionClient = new SubscriptionClient(
+    `ws://localhost:4000/graphql`,
+    {
+      reconnect: true,
+    },
+    ws
+  )
+
+  const client = createClient({
+    fetch,
+    url: GRAPHQL_ENDPOINT,
+    exchanges: [
+      ...defaultExchanges,
+      subscriptionExchange({
+        forwardSubscription(operation) {
+          return subscriptionClient.request(operation)
+        },
+      }),
+    ],
+  })
+
+  const Div = props => (
+    <Box width={80} textWrap="wrap" flexDirection="column" {...props} />
+  )
+
+  const u = unified()
```

Comment:
I think this should be done on a server, not in CLI.

Reply:
answered in comment in main thread